src/query/frontend/ast/cypher_main_visitor.cpp

```diff
@@ -239,6 +239,23 @@ antlrcpp::Any CypherMainVisitor::visitDumpQuery(MemgraphCypher::DumpQueryContext
   return dump_query;
 }
 
+antlrcpp::Any CypherMainVisitor::visitAnalyzeGraphQuery(MemgraphCypher::AnalyzeGraphQueryContext *ctx) {
+  auto *analyze_graph_query = storage_->Create<AnalyzeGraphQuery>();
+  if (ctx->listOfColonSymbolicNames()) {
+    analyze_graph_query->labels_ =
+        std::any_cast<std::vector<std::string>>(ctx->listOfColonSymbolicNames()->accept(this));
+  } else {
+    analyze_graph_query->labels_.emplace_back("*");
+  }
+  if (ctx->DELETE()) {
+    analyze_graph_query->action_ = AnalyzeGraphQuery::Action::DELETE;
+  } else {
+    analyze_graph_query->action_ = AnalyzeGraphQuery::Action::ANALYZE;
+  }
```

Comment:
do handle analyze and throw if both not present

Reply:
we want a good exception if an error happens with syntax here

Reply:
or maybe I'm wrong

Reply:
I understand what you mean but wouldn't agree because there is no way to handle analyze except that is not delete + the exception will be thrown any way.

Reply:
Exception will be thrown if there is something missing, i.e. you write `analyze graph on labels * delete ` without statistics, so I think all good here.